### ISSUE TITLE
Opening a project workspace stuck on spinning pinwheel

### ISSUE THREAD

**Issue by chrowe** [gitbutlerapp/gitbutler#4218]
When trying to apply a remote branch I also get a pinwheel in the `Apply` button.

Version 0.12.7
MacOS 14.5
Apple M1

**Comment by chrowe** [comment]
Not sure if it matters, but this is after an upgrade. I get the same result if I download the latest version or the the nightly though.

**Comment by Byron** [comment]
Thanks for reporting! Could you check [the logs](https://docs.gitbutler.com/development/debugging#logs) to see if there is any obvious `ERROR` lines?

Also, if you wait 60 seconds, does the UI show a "Timeout" popup? This would be an indicator for the backend panicking, which should appear in the `tokio-console` log.

**Comment by chrowe** [comment]
There are no errors in the logs just a bunch of `INFO` lines and it never seems to time out.

**Comment by chrowe** [comment]
Ok, eventually it opened. Not sure how long it took though.
Closed and reopened and it is doing the same thing. Took about 9 minutes.

**Comment by Byron** [comment]
Thanks a lot! Did you find the term `panic` in the `tokio-console` log? Usually when the UI times out, that's the problem. The line would also reveal where it happened, which would help with a fix.
Thanks again

**Comment by chrowe** [comment]
No panic in the tokio-console.

It is also still stuck applying a virtual branch.

**Comment by Byron** [comment]
Yes, and there [is documentation](https://docs.gitbutler.com/troubleshooting/recovering-stuff#gitbutler-references) that should help you to recover your work as well.

I hope that works for you, but if not please come back here and we will do our best to help.

**Comment by krlvi** [comment]
This should be fixed now. If this is still an issue, please feel free to open a new ticket
